### ISSUE TITLE
The Radeon Vega64 can't read the Fan speed by ROCm 1.8.2

### ISSUE THREAD

**Issue by heero-yuy** [RadeonOpenCompute/ROC-smi#36]
Hi,

Checked that the ROCm 1.8.2 on Vega64 FE, found the fan speed is always on 0.0% shown as below:

```
====================    ROCm System Management Interface    ====================
================================================================================
 GPU  Temp    AvgPwr   SCLK     MCLK     Fan      Perf    SCLK OD    MCLK OD
  0   N/A     N/A      N/A      N/A      0%       N/A       N/A        N/A
  1   36c     19.0W    852Mhz   167Mhz   0.0%     auto      0%         0%
================================================================================
====================           End of ROCm SMI Log          ====================
```

But When I set the Fan speed manually by "./rocm-smi --setfan 255" or  "./rocm-smi --setfan 100" or "./rocm-smi --resetfan", the Fan speed on Vega64 is under controlled, I checked that the ROCm 1.8.0 didn't have this problem yet, may I lose something?

OS: Ubuntu 16.04.5 Server
Kernel: 4.4.0-131 & 4.13.0-45

**Comment by jlgreathouse** [comment]
This appears to be an issue with [line 319 of rocm-smi.py](https://github.com/RadeonOpenCompute/ROC-smi/blob/roc-1.8.x/rocm_smi.py#L319). The goal of this function is to get a percentage of maximum fan speed, so fanLevel is always <= fanMax. However, dividing two integers ([outside of Python 3](https://stackoverflow.com/questions/183853/in-python-2-what-is-the-difference-between-and-when-used-for-division)) is going to `floor()` the result, leading to this function always returning either 1 (at max fan speed) or 0 (at anything else).

Thank you for reporting this.

**Comment by kentrussell** [comment]
Thanks for that. I've got a fix in for the 1.9 branch (and a ton of other fixes too), it's just waiting to be released with the rest of ROCm. I'll close this once 1.9 gets pushed.

**Comment by kentrussell** [comment]
I just pushed the fixes to the krussell/fixes branch. Can you give that a shot and let me know if it works? I don't know when 1.9 will be released so this will hopefully help in the meantime

**Comment by kentrussell** [comment]
Closing as fix is available on krussell/fixes, roc-1.9.x branch and in the SMI package (released tomorrow on the apt server)

**Comment by heero-yuy** [comment]
Hi All,
Checked that the ROCm 1.9.0 installed on Ubuntu 18.04.1 is fixed, thanks!
